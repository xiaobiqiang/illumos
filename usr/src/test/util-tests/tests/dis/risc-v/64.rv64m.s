/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2018, Joyent, Inc.
 */

/*
 * Test our disassembly of the RV64M instructions. Instructions are ordered per the
 * ISA manual.
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	mulw	s0, s1, s2
	divw	s4, s5, s6
	divuw	s5, s6, s7
	remw	s6, s7, s8
	remuw	s7, s8, s9
.size libdis_test, [.-libdis_test]
