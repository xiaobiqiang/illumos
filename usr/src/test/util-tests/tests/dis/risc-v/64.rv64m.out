    libdis_test:      3b 84 24 03        mulw s0,s1,s2
    libdis_test+0x4:  3b ca 6a 03        divw s4,s5,s6
    libdis_test+0x8:  bb 5a 7b 03        divuw s5,s6,s7
    libdis_test+0xc:  3b eb 8b 03        remw s6,s7,s8
    libdis_test+0x10: bb 7b 9c 03        remuw s7,s8,s9
